### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by 187jjay187**
### Blog app: creating a data model
In this project, I used migration files to build my app schema.

**These are the project requirements:**

- Create a new app. Give it a meaningful name
- Initialize your project with Git
- Make sure that your project has Postgres database set up
- Build your project schema
- Create and run the necessary migration files
- Table and column names should match the ERD diagram
- Foreign keys should be included.
- All columns that are foreign keys should have a corresponding index

### CODE REVIEW

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @187jjay187 , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please when trying to run your migrations, I get this error. From what I realized in your files, the user migration is created at last which is not very good because the order in which the tables are being created matters. Ill recommend creating the tables first before creating the relations
image

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by 187jjay187** [comment]
Hey there @m05e5 please could you assist as I either don't understand the required changes clearly or everything seems to be working on my side
bandicam 2023-03-04 04-47-01-325

**Comment by BenMukebo** [comment]
Hello @187jjay187, I'm getting the same error message 

image

- Could you please reclone the project and run the migration?

**Comment by 187jjay187** [comment]
Hello @BenMukebo i've deleted the old repo and cloned it again from my most recent github link, and I am still not seeing the error after running the command multiple times

bandicam 2023-03-04 05-36-17-904

File: db/migrate/20230303192614_add_user_ref_to_likes.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+    def change
+      add_reference :likes, :author, null: false, foreign_key: { to_table: 'users' }
+    end
+  end
```
Comment (BenMukebo) [inline db/migrate/20230303192614_add_user_ref_to_likes.rb:5]: - [OPTIONAL] Hello @187jjay187, everything looks fine However as requested by the previous reviouwer, This error message indicates that there was an error while running a database migration, which caused the migration - process to be aborted.
Here are some steps you can take to diagnose and fix the issue:
  
1. Check the database schema: Verify that the database schema matches the expected schema defined in your migration files.
2. Roll back the migration: If you're still having trouble, you may need to roll back the migration and try again. You can do this by running the rails db:rollback command.

**Review by BenMukebo — APPROVED** [review]
### Hi @187jjay187, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- Database adapter is using Postgresql. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230303192614_add_user_ref_to_likes.rb (hunk 1 above)
Comment (187jjay187) [reply db/migrate/20230303192614_add_user_ref_to_likes.rb:5]: @BenMukebo thank you so much for the help and directions I really do appreciate it and will definitely look into it before trying to move on

**Pull request merged**